### ISSUE TITLE
a couple of notes about deb creation script

### ISSUE THREAD

**Issue by claurita** [OpenBoard-org/OpenBoard#134]
Hi again.
A little remark on this too.
I noticed that in the script package.sh, you used apt-cache show to extract the versions of the dependencies for the deb package:

echo -n "${tab[$i]} (>= "`apt-cache show ${tab[$i]} | grep "Version: " | head -1 | awk '{      print $2 }' | sed -e 's/\([:. 0-9?]*\).*/\1/g' | sed -e 's/\.$//'`") " >> "$CONTROL_FILE"

Please, consider that this could lead to problems if you made an update to the cache (e.g. by apt-get update) without actually updating all the packages involved in OB building. The version you get this way, refer to the packages in repositories, not those effectively used to build.
I know that it's a tricky situation, but anybody using this script is much probably building for his own machine, not to pull the package in the official repositories. And often the sid repositories are in an "unstable" (indeed it's called unstable) state so you cannot (or shouldn't) do a full upgrade.

More, you get only the relevant part of the version string (filtering by sed). I totally agree, but this could lead to problems with dpkg (even if I consider this a sort of bug in dpkg):
e.g. consider the current (in sid) version of libqt5webkit5:
5.212.0\~alpha2-9
If you truncate the version to 5.212.0 and build the control file with >=5.212.0, dpkg refuses to install considering 5.212.0\~alpha2-9 not yet a "real" 5.212.0. This could be theoretically fine, but ends up with an not installable package.
In conclusion, I had to lazily change the script with:

echo -n "${tab[$i]} (>= "`dpkg -s ${tab[$i]} | grep "Version: " | head -1 | awk '{  print $2 }'`") " >> "$CONTROL_FILE"

Again, I wonder your thoughts about these notes.
Claudio

**Comment by juanjosepablos** [comment]
Claudio, 
please give some feedback on #170 to see if it gets resolved.
